src/logcollector/read_journald.c

```diff
@@ -0,0 +1,64 @@
+#include "shared.h"
+#include "logcollector.h"
+#include <systemd/sd-journal.h>
+
+void prime_sd_journal(sd_journal **jrn) {
+  int ret;
+  ret = sd_journal_open(&jrn, SD_JOURNAL_LOCAL_ONLY);
+  ret = sd_journal_get_fd(jrn);
+  ret = sd_journal_seek_tail(jrn);
+  // prime sd_journal_next before the reader loop
+  ret = sd_journal_previous(jrn);
+}
+
+void sd_read_journal(char *journal) {
+  sd_journal *jrn;
+  int ret;
+  const char *data;
+  size_t len;
+  struct timeval tv;
+  uint64_t curr_timestamp;
+  time_t nowtime;
+  struct tm *nowtm;
+  char tmbuf[64], final_msg[OS_MAXSTR];
+
+  prime_sd_journal(&jrn);
+  ret = sd_journal_next(j);
```

Comment:
Should be `et = sd_journal_next(jrn);`

Reply:
This is why professional developers sleep before coding, ty, will fix.